jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```
@@ -0,0 +1,163 @@
+        String sessionConfigs = dbAndSessionConfs.substring(dbName.length() + 1);
+        String[] params = sessionConfigs.split(";");
+
+        StringBuilder otherSessionConfsBuilder = new StringBuilder();
+        String projectId = null;
+        String region = null;
+        String clusterPoolLabel = null;
+        String clusterName = null;
+        int port = HIVE_DEFAULT_PORT;
+        for (String paramPair : params) {
+            if (!paramPair.isEmpty() && paramPair.indexOf("=") != -1) {
+                String field = paramPair.split("=")[0];
+                String value = paramPair.split("=")[1];
+
+                switch (field) {
+                    case "clusterName":
+                        clusterName = value;
+                        break;
+                    // For not yet implemented submitting to cluster Pool.
+                    // Optional if clusterName is given.
+                    case "projectId":
+                        projectId = value;
+                        break;
+                    case "region":
+                        region = value;
+                        break;
+                    case "clusterPoolLabel":
+                        clusterPoolLabel = value;
+                        break;
+                    // optional field
+                    case "port":
+                        try {
+                            port = Integer.valueOf(value);
+                        } catch (NumberFormatException e) {
+                            throw new InvalidURLException(
+                                    String.format(
+                                            "Error: port=%s \n"
+                                                    + "Please indicate correct port number or remove the field.",
+                                            value));
+                        }
+                        break;
+                    // No need to parse value, already set in HiveJdbcParams
```

I think it should be an error to set these values. Silently ignoring them is a bit confusing

---

Since we provide these two values as default, if the user specify them again in the url, probably we could ignore them, and only when they specify the wrong values do we throw an exception? (originally I was thinking that even if they specify `transportMode` and `httpPath` wrongly, we still redirect the corrected setting connection to Hive and let `HiveConnection` decide what to do) Let me know what you think, thanks.